core/src/main/scala/org/apache/spark/status/api/v1/ApiRootResource.scala

```diff
@@ -148,38 +148,36 @@ private[v1] trait BaseAppResource extends ApiRequestContext {
 }
 
 private[v1] class ForbiddenException(msg: String) extends WebApplicationException(
-  Response.status(Response.Status.FORBIDDEN).entity(msg).build())
+  Response.status(Response.Status.FORBIDDEN).entity(msg).`type`(MediaType.TEXT_PLAIN).build())
```

Comment:
You can extract a new helper function to build the response object from a `Response.Status` and a message. I think UIUtils is good place for such a function then you can use it at `OneApplicationResource.scala` too as the `serverError`method just sets the status to `Status.INTERNAL_SERVER_ERROR` which overwritten by right away with `Response.Status.SERVICE_UNAVAILABLE`.